attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -200,16 +191,20 @@ impl AttestationAPIs for AttestationAgent {
             pcr
         });
 
-        let log_entry = EventEntry::new(domain, operation, content);
-        let event_digest = log_entry.digest_with(self.config.eventlog_config.eventlog_algorithm);
+        let content: Content = content.try_into()?;
 
-        let mut eventlog = self.eventlog.lock().await;
+        let log_entry = LogEntry::Event {
+            domain,
+            operation,
+            content,
+        };
+        let event_digest = log_entry.digest_with(self.config.eventlog_config.eventlog_algorithm);
 
         self.attester
             .extend_runtime_measurement(event_digest, register_index)
             .await?;
 
-        eventlog.write_log(&log_entry.to_string())?;
+        self.eventlog.lock().await.write_log(&log_entry)?;
```

Comment:
There's a very subtle synchronization issue here. 

Line 207 makes the lock only owned by the execution stream this line, while the original the whole `async fn extend_runtime_measurement()` function body owns the lock.

We'd better keep the lock being owned before and after `self.attester.extend_runtime_measurement()` calling to make `self.attester.extend_runtime_measurement()` and `write_log()` an atomic operation. without atomic semantics, when multiple `extend_runtime_measurement` requests come together, `write_log()` might not happen right after its `self.attester.extend_runtime_measurement()` but in out of order.

Reply:
ah, good observation. I thought the lock was mostly about encapsulating the sync io from the async wrapper. I put it into a block to make that more obvious